### ISSUE TITLE
JSCS - checker javascript/jscs: unrecognized error format

### ISSUE THREAD

**Issue by thomb** [scrooloose/syntastic#1682]
I've checked the other reports regarding this issue, and I've had no success with some of the suggested workarounds, though I believe I have found the root cause of the issue.

By default JSCS limits the number of errors shown, and when running it on code which /really/ breaks the standards which are set (i.e. errors greater than maxErrors) JSCS returns non-standard JSON which fails to be parsed.

For example:
```
'{"js/layout.js":[{"line":***,"column":3,"message":"Invalid jsdoc-block definition"}, 
[...] 
{"line":***,"column":3,"message":"Invalid jsdoc-block definition"},]}
Too many errors... Increase `maxErrors` configuration option value to see more.'
```

It's that `maxErrors` message.

I'm not sure if it makes any sense for syntastic to clean up the output of JSCS, or is JSCS should provide an option to suppress this message so valid JSON is always returned.

In any event, this can be worked around by setting 
```
"maxErrors": -1
```
in your .jscsrc.

Mainly reporting here in case other people run into this.

**Comment by lcd047** [comment]
> I've checked the other reports regarding this issue

Please show any previous reports of this issue.

> By default JSCS limits the number of errors shown, and when running it on code which /really/ breaks the standards which are set (i.e. errors greater than maxErrors) JSCS returns non-standard JSON which fails to be parsed.

Fixed in de0106e.

**Comment by thomb** [comment]
That statement was not clear, I was referring to previous reports of "checker javascript/jscs: unrecognized error format".

Thanks for the fix.
